src/main/kotlin/calculator/Parser.kt

```diff
@@ -0,0 +1,28 @@
+package calculator
+
+import calculator.common.isNotNumber
+import calculator.common.isNotOperator
+
+class Parser(private val limit: Int) {
+
+    fun parse(input: String?): List<String> {
+        if (input.isNullOrBlank()) {
+            return emptyList()
+        }
+        return input.split(" ")
+    }
+
+    @Throws(IllegalArgumentException::class)
+    fun validate(elements: List<String>): Boolean {
+        if (elements.isEmpty()) throw IllegalArgumentException("elements is empty")
+        if (elements.size < limit) throw IllegalArgumentException("elements size less than min size")
```

Comment:
코틀린에서는 `IllegalArgumentException` 을 아래와 같은 방법으로도 발생시킬 수 있습니다.
(또한 개수에 대한 검증이 비어있는지를 함께 검사해주겠네요!)

```suggestion
    fun validate(elements: List<String>): Boolean {
        require(elements.size >= limit) { "elements size less min size" }
```

Reply:
`require` 외에도 `requireNotNull`, `check`, `checkNotNull` 등이 있습니다.

Reply:
아! 이게 있었네요. 😄 수정하여 반영합니다.